Demos/benchmarks/run_benchmarks.py

```diff
@@ -0,0 +1,146 @@
+import collections
+import logging
+import pathlib
+import shutil
+import subprocess
+import sys
+import tempfile
+
+
+BENCHMARKS_DIR = pathlib.Path(__file__).parent
+
+BENCHMARK_FILES = sorted(BENCHMARKS_DIR.glob("bm_*.py"))
+
+ALL_BENCHMARKS = [bm.stem for bm in BENCHMARK_FILES]
+
+
+def run(command, cwd=None):
+    try:
+        return subprocess.run(command, cwd=str(cwd) if cwd else None, check=True, capture_output=True)
+    except subprocess.CalledProcessError as exc:
+        logging.error(f"Command failed: {' '.join(map(str, command))}\nOutput:\n{exc.stderr.decode()}")
+        raise
+
+
+def compile_benchmarks(cython_dir: pathlib.Path, bm_dir: pathlib.Path, benchmarks=None, cythonize_args=None):
+    cythonize_args = cythonize_args or []
+
+    util_file = BENCHMARKS_DIR / "util.py"
+    if util_file.exists():
+        shutil.copy(util_file, bm_dir / util_file.name)
+
+    bm_files = []
+    for bm_src_file in BENCHMARK_FILES:
+        if benchmarks and bm_src_file.stem not in benchmarks:
+            continue
+        bm_file = bm_dir / bm_src_file.name
+        shutil.copy(bm_src_file, bm_file)
+        for dep in BENCHMARKS_DIR.glob(bm_src_file.stem + ".pxd"):
+            shutil.copy(dep, bm_dir / dep.name)
+        bm_files.append(bm_file)
+
+    if bm_files:
+        bm_count = len(bm_files)
+        logging.info(f"Compiling {bm_count} benchmark{'s' if bm_count != 1 else ''}: {', '.join(bm_file.stem for bm_file in bm_files)}")
+        run([sys.executable, str(cython_dir / "cythonize.py"), f"-j{bm_count or 1}", "-i", *bm_files, *cythonize_args], cwd=cython_dir)
+
+
+def git_clone(rev_dir, revision):
+    run(["git", "clone", "-n", "--no-single-branch", ".", str(rev_dir)])
+    run(["git", "switch", "-C", revision, f"refs/remotes/origin/{revision}"], cwd=rev_dir)
```

Comment:
For some reason that I don't understand, this fails to find the `master` branch how ever I tried.